### PR TITLE
Add ShardedDaemonProcess integration

### PR DESCRIPTION
**Pull request by Arkatufus**
## Changes

* Bump Akka to 1.5.24
* Add integration for ShardedDaemonProcess and ShardedDaemonProcess proxy

### CODE REVIEW

File: src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs [hunk 1]
```diff
@@ -1011,6 +1011,69 @@ async Task Resolver(ActorSystem system, IActorRegistry registry, IDependencyReso
                 registry.Register<TKey>(shardRegionProxy);
             });
         }
+    
+        public static AkkaConfigurationBuilder WithShardedDaemonProcess<TKey>(
```
Comment (Aaronontheweb) [inline src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs:1046]: LGTM - pretty straightforward

**Review by Aaronontheweb — APPROVED** [review]
LGTM - looks pretty clear to me

File: src/Akka.Hosting.TestKit/TestKit.cs [hunk 2]
```diff
@@ -102,20 +102,25 @@ private void InternalConfigureServices(HostBuilderContext context, IServiceColle
...
                         base.InitializeTest(system, (ActorSystemSetup)null!, null, null);
-                        _initialized.SetResult(Done.Instance);
+                        registry.Register<TestProbe>(TestActor);
                     }
                     catch (Exception e)
                     {
                         _initialized.SetException(e);
                     }
                 });
+
```
Comment (Arkatufus) [inline src/Akka.Hosting.TestKit/TestKit.cs:117]: Need to move initialization to the start of the chain to make sure that we can inject `TestActor` into test actor constructors

File: src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs [hunk 3]
```diff
@@ -1011,6 +1012,127 @@ async Task Resolver(ActorSystem system, IActorRegistry registry, IDependencyReso
...
+        ///     The type key to use to retrieve the <see cref="IActorRef"/> for this <see cref="DaemonMessageRouter"/>.
+        /// </typeparam>
+        /// <returns>
+        ///     The same <see cref="AkkaConfigurationBuilder"/> instance originally passed in.
+        /// </returns>
+        public static AkkaConfigurationBuilder WithShardedDaemonProcess<TKey>(
+            this AkkaConfigurationBuilder builder,
+            string name,
+            int numberOfInstances,
+            Func<ActorSystem, IActorRegistry, IDependencyResolver, Func<int, Props>> entityPropsFactory,
```
Comment (Arkatufus) [inline src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs:1050]: Need to make the Props factory like this to support dependency injection

**Review by Arkatufus — COMMENTED** [review]
Self review

File: src/Akka.Cluster.Hosting.Tests/ShardedDaemonProcessProxySpecs.cs
Comment (Aaronontheweb) [inline src/Akka.Cluster.Hosting.Tests/ShardedDaemonProcessProxySpecs.cs:1]: lgtm

File: src/Akka.Hosting.TestKit/TestKit.cs [hunk 4]
```diff
@@ -102,20 +102,25 @@ private void InternalConfigureServices(HostBuilderContext context, IServiceColle
...
-                ConfigureAkka(builder, provider);
-
-                builder.AddStartup((system, _) =>
+                builder.StartActors((system, registry) =>
                 {
                     try
                     {
                         base.InitializeTest(system, (ActorSystemSetup)null!, null, null);
-                        _initialized.SetResult(Done.Instance);
+                        registry.Register<TestProbe>(TestActor);
```
Comment (Aaronontheweb) [inline src/Akka.Hosting.TestKit/TestKit.cs:110]: Ah, interesting

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Pull request merged**